Explorer/Assets/Scripts/Global/Dynamic/PortableExperiences/PortableExperiencesController.cs

```diff
@@ -0,0 +1,218 @@
+using System.Linq;
+using GetSceneDefinition = ECS.SceneLifeCycle.SceneDefinition.GetSceneDefinition;
+
+namespace PortableExperiences.Controller
+{
+    public class PortableExperiencesController : IPortableExperiencesController
+    {
+        private static readonly QueryDescription GET_SCENE_DEFINITION = new QueryDescription().WithAll<GetSceneDefinition>();
+        private static readonly QueryDescription SCENE_DEFINITION_COMPONENT = new QueryDescription().WithAll<SceneDefinitionComponent>();
+
+        private readonly ServerAbout serverAbout = new ();
+        private readonly IWeb3IdentityCache web3IdentityCache;
+        private readonly IWebRequestController webRequestController;
+        private readonly IScenesCache scenesCache;
+
+        private readonly RealmData realmData = new ();
+
+        private readonly ObjectProxy<World> globalWorldProxy;
+        private List<Entity> entitiesToDestroy = new ();
+        public Dictionary<string, Entity> PortableExperienceEntities { get; } = new ();
+        private World world => globalWorldProxy.Object;
+        private List<IPortableExperiencesApi.SpawnResponse> spawnResponsesList = new ();
+
+        public PortableExperiencesController(
+            ObjectProxy<World> world,
+            IWeb3IdentityCache web3IdentityCache,
+            IWebRequestController webRequestController,
+            IScenesCache scenesCache)
+        {
+            globalWorldProxy = world;
+            this.web3IdentityCache = web3IdentityCache;
+            this.webRequestController = webRequestController;
+            this.scenesCache = scenesCache;
+        }
+
+        public async UniTask<IPortableExperiencesApi.SpawnResponse> CreatePortableExperienceAsync(string ens, string urn, CancellationToken ct, bool isGlobalPortableExperience = false)
+        {
+            //According to kernel implementation, the id value is used as an urn
+            //https://github.com/decentraland/unity-renderer/blob/b3b170e404ec43bb8bc08ec1f6072812005ebad3/browser-interface/packages/shared/apis/host/PortableExperiences.ts#L28
+            //And is validated first. As it has no ipfs config, it uses the one from the current realm apparently
+
+            string worldUrl = string.Empty;
+
+            if (!string.IsNullOrEmpty(urn))
+            {
+                //TODO: Enable loading PX from urns using current scene realm data. -> will be done in next iteration.
+                //worldUrl = IpfsHelper.ParseUrn(urn).BaseUrl.Value;
+            }
+
+            if (EnsUtils.ValidateEns(ens)) { worldUrl = EnsUtils.ConvertEnsToWorldUrl(ens); }
+
+            if (!worldUrl.IsValidUrl()) throw new ArgumentException("Invalid Spawn params. Provide a URN or an ENS name");
+
+            var portableExperiencePath = URLDomain.FromString(worldUrl);
+            URLAddress url = portableExperiencePath.Append(new URLPath("/about"));
+
+            await UniTask.SwitchToMainThread();
```

Comment:
Why are we switching to main thread here? If this is called from the js context, then i think the switch should be done at `PortableExperiencesApiWrapper`

Reply:
Fixed it, now its called from the API Implementation actually!